mapreduce/metrics.csv

```diff
@@ -0,0 +1,27 @@
+mapreduce.job.reduces_total,histogram,,reduces,,The total number of reduces,0,mapreduce,mr_job_reduces
+mapreduce.job.reduces_completed,histogram,,reduces,,The number of completed reduces,0,mapreduce,mr_job_reduces_completed
+mapreduce.job.maps_pending,histogram,,maps,,The number of maps still to be run,0,mapreduce,mr_job_maps_pending
+mapreduce.job.maps_running,histogram,,maps,,The number of running maps,0,mapreduce,mr_job_maps_running
+mapreduce.job.reduces_pending,histogram,,reduces,,The number of reduces still to be run,0,mapreduce,mr_job_reduces_pending
+mapreduce.job.reduces_running,histogram,,reduces,,The number of running reduces,0,mapreduce,mr_job_reduces_running
+mapreduce.job.new_reduce_attempts,histogram,,reduces,,The number of new reduce attempts,0,mapreduce,mr_job_new_reduce_attempts
+mapreduce.job.running_reduce_attempts,histogram,,reduces,,The number of running reduce attempts,0,mapreduce,mr_job_running_reduce_attempts
+mapreduce.job.failed_reduce_attempts,histogram,,reduces,,The number of failed reduce attempts,0,mapreduce,mr_job_failed_reduce_attempts
+mapreduce.job.killed_reduce_attempts,histogram,,reduces,,The number of killed reduce attempts,0,mapreduce,mr_job_killed_reduce_attempts
+mapreduce.job.successful_reduce_attempts,histogram,,reduces,,The number of successful reduce attempts,0,mapreduce,mr_job_successful_reduce_attempts
+mapreduce.job.new_map_attempts,histogram,,maps,,The number of new map attempts,0,mapreduce,mr_job_new_map_attempts
+mapreduce.job.running_map_attempts,histogram,,maps,,The number of running map attempts,0,mapreduce,mr_job_running_map_attempts
+mapreduce.job.failed_map_attempts,histogram,,maps,,The number of failed map attempts,0,mapreduce,mr_job_failed_map_attempts
+mapreduce.job.killed_map_attempts,histogram,,maps,,The number of killed map attempts,0,mapreduce,mr_job_killed_map_attempts
+mapreduce.job.successful_map_attempts,histogram,,maps,,The number of successful map attempts,0,mapreduce,mr_job_successful_map_attempts
+mapreduce.job.counter.reduce_counter_value,histogram,,,,The counter value of reduce tasks,0,mapreduce,mr_job_reduce_counter
+mapreduce.job.counter.map_counter_value,histogram,,,,The counter value of map tasks,0,mapreduce,mr_job_map_counter
+mapreduce.job.counter.total_counter_value,histogram,,,,The counter value of all tasks,0,mapreduce,mr_job_total_counter
+mapreduce.job.map.task.progress,histogram,,progress,,The distribution of all map task progresses,0,mapreduce,mr_map_task_progress
+mapreduce.job.map.task.elapsed_time,histogram,,ms,,The distribution of all map tasks elapsed time (in ms),0,mapreduce,mr_map_task_time
+mapreduce.job.reduce.task.progress,histogram,,progress,,The distribution of all reduce task progresses,0,mapreduce,mr_map_task_progress
+mapreduce.job.reduce.task.elapsed_time,histogram,,ms,,The distribution of all reduce tasks elapsed time (in ms),0,mapreduce,mr_map_task_time
```

Comment:
Unfortunately the `histogram` metric type doesn't exist for the metrics definitions. Each `histogram` metric is actually sent as 5 different metrics:

  name | type
  -----|------------
  metric.max | `gauge`
  metric.avg | `gauge`
  metric.median | `gauge`
  metric.95percentile | `gauge`
  metric.count | `gauge`

So for each histogram metric that the check collects you have to replace that metric with the 5 metrics that are sent.

Tell me if something's not clear.